src/components/include/utils/lock.h

```diff
@@ -61,7 +63,11 @@ class SpinMutex {
       return;
     }
     for(;;) {
+#if defined(_MSC_VER)
```

Comment:
It seems wrong to use at the same time MS defines and own define
Need to used one of them, i think